src/Playwright.NUnit/TimeoutCommand.cs

```diff
@@ -0,0 +1,129 @@
+using System;
+using System.Diagnostics;
+using System.Threading.Tasks;
+using Microsoft.Playwright.TestAdapter;
+using NUnit.Framework;
+using NUnit.Framework.Interfaces;
+using NUnit.Framework.Internal;
+using NUnit.Framework.Internal.Commands;
+
+namespace Microsoft.Playwright.NUnit;
+
+public class TimeoutCommand : BeforeTestCommand
+{
+    protected bool TearDownRan { get; set; }
+    internal TimeoutCommand(TestCommand innerCommand) : base(innerCommand) { }
+
+    public override TestResult Execute(TestExecutionContext context)
+    {
+        var timeout = PlaywrightSettingsProvider.TestTimeout;
+        if (timeout == null || Debugger.IsAttached)
+        {
+            return innerCommand.Execute(context);
+        }
+
+        if (PlaywrightTimeoutIsSmallerThanNUnitTimeout(context, timeout.Value, out var result))
+        {
+            return result!;
+        };
+
+        ExecuteWithTimeoutAsync(context, timeout.Value).GetAwaiter().GetResult();
+        return context.CurrentResult;
+    }
+
+    private async Task ExecuteWithTimeoutAsync(TestExecutionContext context, int timeout)
```

Comment:
This needs a comment that explains whether this function runs setups and teardowns.